src/static/js/chapter.js

```diff
@@ -351,7 +351,7 @@ function indexHighlighter() {
   }, options);
 
   // Add an intersection observer to each heading
-  var all_headings = document.querySelectorAll('article h1, article h2, article h3');
+  var all_headings = document.querySelectorAll('article h1, article h2, article h3, article h4');

```

Comment:
As suggested let's future proof this:

```suggestion
  var all_headings = document.querySelectorAll('article h1, article h2, article h3, article h4, article h5, article h6');
```

Reply:
I think that's a good idea.